pkg/leader/leader_test.go

```diff
@@ -0,0 +1,149 @@
+package leader
+
+import (
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+func Test_computeConfig(t *testing.T) {
+	type args struct {
+		rl  resourcelock.Interface
+		cbs leaderelection.LeaderCallbacks
+	}
+	type env struct {
+		key   string
+		value string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		envs    []env
+		want    *leaderelection.LeaderElectionConfig
+		wantErr bool
+	}{
+		{
+			name: "all defaults",
```

Comment:
This test and some of the others are dependent on running in a specific order since other tests may change the default env. If you create a resetEnv function that is run before every test and calls `os.Unsetenv` on all desired variables then we can have more deterministic results.